packages/flutter/lib/src/foundation/platform.dart

```diff
@@ -94,5 +97,17 @@ enum TargetPlatform {
 /// certainly widgets to work assuming the presence of a system-wide back
 /// button, which will make those widgets unusable since iOS has no such button.
 ///
-/// In general, therefore, this property should not be used in release builds.
-TargetPlatform? debugDefaultTargetPlatformOverride;
+/// Attempting to override this property in non-debug builds causes an error.
+TargetPlatform? get debugDefaultTargetPlatformOverride =>
+    _debugDefaultTargetPlatformOverride;
+
+set debugDefaultTargetPlatformOverride(TargetPlatform? value) {
+  if (!kDebugMode) {
+    throw FlutterError(
+      'Cannot modify debugDefaultTargetPlatformOverride in non-debug builds.\n'
```

Comment:
There's a stray `\n` at the end of the message.

Reply:
Thanks, done!